### PR TITLE
feat: keep attachments array and remove file and image uploads in MessageInput state

### PR DESCRIPTION
**Pull request by MartinCupela**
### 🎯 Goal

Remove the following state variables from the MessageInput state:

- `fileOrder`
- `imageOrder`
- `fileUploads`
- `imageUploads`

All the attachments except the link previews (attachment contains `og_scrape_url` attribute) are now stored in `attachments` array. Array keeps order and the attachment type can be determined by `attachment.type` property.

Default attachment types are:

- `audio`
- `file`
- `image`
- `video`
- `voiceRecording`

Uploaded attachments will be determined by `file` attribute as `attachment.localMetadata.file`.

Remove the following API from the MessageInput state
- `uploadFile` -> from now on, use `uploadAttachment`
- `uploadImage` -> from now on, use `uploadAttachment`
- `removeFile` -> from now on, use `removeAttachments`
- `removeImage` -> from now on, use `removeAttachments`

The function `uploadNewFiles` has been kept.

So to sum up, the attachments management API in MessageInput is from now on only:

- `attachments`
- `uploadAttachment`
- `uploadNewFiles`
- `upsertAttachments`
- `removeAttachments`

### 🎨 UI Changes

No UI changes.

BREAKING CHANGE:  Removed fileOrder, imageOrder, fileUploads, imageUploads, uploadFile, uploadImage, removeFile, removeImage from the MessageInputContext.

### CODE REVIEW

File: docusaurus/docs/React/release-guides/upgrade-to-v12.mdx [hunk 1]
```diff
@@ -4,6 +4,65 @@ title: Upgrade to v12
...
+- `fileOrder` - An array of IDs for non-image uploaded attachments. Its purpose was just to keep order in which the attachments were added.
+- `imageOrder` - An array of IDs for image uploaded attachments. Its purpose was just to keep order in which the attachments were added.
+- `fileUploads` - A mapping of ID to attachment representing the uploaded non-image file.
+- `imageUploads` - A mapping of ID to attachment representing the uploaded image file.
+
+These four variables' purpose was to render the attachment previews in a given order. All the non-image attachments were assigned type `"file"` until the message was sent at which point the audio files were assigned type `"audio"` and video file type `"video"`. Also, the attachment objects structure used to change upon submission. Some keys were renamed and other just removed.
+
+The shortcomings of the above approach have been adjusted as follows:
+
+1. All the attachments except the link preview attachments (objects that contain `og_scrape_url` attribute) are now stored in `attachments` array of `MessageInputContext`. Array keeps order and the attachment type can be determined by the `attachment.type` property.
```
Comment (MartinCupela) [inline docusaurus/docs/React/release-guides/upgrade-to-v12.mdx:20]: I am not sure if we should not include the link previews in `attachments` too.

**Review by oliverlaz — APPROVED** [review]
